nailgun/entities.py

```diff
@@ -8132,3 +8132,149 @@ def __init__(self, server_config=None, **kwargs):
         }
         self._meta = {'api_path': 'katello/api/v2/srpms'}
         super().__init__(server_config, **kwargs)
+
+
+class Webhook(
+    Entity,
+    EntityCreateMixin,
+    EntityDeleteMixin,
+    EntityReadMixin,
+    EntitySearchMixin,
+    EntityUpdateMixin,
+):
+    """A representation of a Webhook entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'name': entity_fields.StringField(
+                required=True, str_type='alpha', length=(6, 12), unique=True
+            ),
+            'enabled': entity_fields.BooleanField(),
+            'event': entity_fields.StringField(
+                required=True,
+                choices=[
```

Comment:
That's certainly something we can think about in the future. It would be a bit less efficient but might help with maintainability.  Admittedly, it would be pretty low priority.